src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java

```diff
@@ -304,7 +304,10 @@ public class XdocsPagesTest {
      */
     @BeforeAll
     public static void generateXdocContent() throws Exception {
+        final String originalLineSeperator = System.getProperty("line.separator");
+        System.setProperty("line.separator", "\n");
         XdocGenerator.generateXdocContent();
+        System.setProperty("line.separator", originalLineSeperator);
```

Comment:
Too hacky.

Let's update generator to use \n instead of system variable.

Reply:
Please make commit that fix generator reference #8884 in commit message. As you fixing real unrelated but blocking issue

Reply:
yep! I know it's hacky, 
but do you want me to raise a separate PR for #8884 to use this temporary fix OR do you want me to mention #8884 in a commit msg in this PR?

Reply:
> Too hacky.
> 
> Let's update generator to use \n instead of system variable.

Ok got it. I am looking out some other good way to fix this!

Reply:
@romani. I had a very deep dive within the code and after exploring a lot of things. These are my findings.

checkstyle is using the `org.​apache.​maven.​doxia.​sink` library for dealing with **xml**. Now there is a default `unifyEOLs(String)` function which ensure that all EOL are strictly same as `System EOL`. So, till we are using this **doxia**, it seems impossible to customize the EOL as long as unifyEOLs is enforcing the **system** EOLs.

Take a look at the code below: `org.​apache.​maven.​doxia.​sink.XhtmlBaseSink`

```
 /** {@inheritDoc} */
    protected void write( String text )
    {
        if ( !this.tableCaptionXMLWriterStack.isEmpty() && this.tableCaptionXMLWriterStack.getLast() != null )
        {
            this.tableCaptionXMLWriterStack.getLast().writeMarkup( unifyEOLs( text ) );
        }
        else if ( !this.tableContentWriterStack.isEmpty() && this.tableContentWriterStack.getLast() != null )
        {
            this.tableContentWriterStack.getLast().write( unifyEOLs( text ) );
        }
        else
        {
            writer.write( unifyEOLs( text ) );
        }
    }
```

I am feeling that we need an alternative to Sink Doxia API to tackle the custom EOL. Although there are plans of removing the module as per #13426.

Let me know, how you would like to proceed on this problem?

Reply:
I do not see also any way to make it configurable.

@kalpadiptyaroy , please confirm that you do not see changes in "meta" files as it is claimed at https://github.com/checkstyle/checkstyle/issues/8884 .

and we have problems only xdoc files, where we definitely use doxia. So atleast we know that problems are different.

@kalpadiptyaroy , is there any way to override `protected void write( String text )` ?
@stoyanK7 , can you help with this ? you spend a lot of time with doxia recently.

Reply:
We have factory and Sink class specific for us - https://github.com/checkstyle/checkstyle/tree/master/src/main/java/com/puppycrawl/tools/checkstyle/site

Reply:
@kalpadiptyaroy 
> `Sink Doxia API ... Although there are plans of removing the module as per https://github.com/checkstyle/checkstyle/issues/13426.`

I think you may be mistaken that doxia will be removed. The issue is moving the doxia from the main/test phase and move it to the site phase, so it is not being released as part of our production or test code.

@romani 
> We have factory and Sink class specific for us 
> can we override this write method at https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/site/XdocsTemplateSink.java

Correct me if I am wrong, but this class is just to have our custom xdoc template files. It is planned to be removed as part of the #13426 at which time we will be using pure doxia classes.

Reply:
@kalpadiptyaroy @romani If this is a sticking point for this PR, I recommend we restore the problem changes before executing this git diff and move this to one of the identified issues mentioned in this PR and add much more documentation of the issue and the fix needed.

Example code we used in eclipse-cs because the timestamp of the file is always modified on every run:
https://github.com/checkstyle/eclipse-cs/blob/ae5102c12468a60055cbee2bd7dda876e81d40f1/.ci/validation.sh#L10-L12

Reply:
> I do not see also any way to make it configurable.
> 
> @kalpadiptyaroy , please confirm that you do not see changes in "meta" files as it is claimed at #8884 .
> 
Yes, as per these results I can assert that we do not see any changes related to **meta** files as claimed in #8884.
https://ci.appveyor.com/project/Checkstyle/checkstyle/builds/48008333/job/0na53h4c1fq203cc#L1870

Reply:
> @kalpadiptyaroy , is there any way to override `protected void write( String text )` ?

The inheritance hierarchy is as follows.

`XdocsTemplateSink` extends `XhtmlBaseSink` 
and `XhtmlBaseSink` extends `AbstractXmlSink`.

Now `write` func. belongs to `AbstractXmlSink` **originally** and `XhtmlBaseSink` **already overrides** it as I have qouted earlier. So it is not possible for `XdocsTemplateSink` to directly override the original func.

And if I override `XhtmlBaseSink`'s `write` func. then I won't be able to access the `writer` variable.

Reply:
> Correct me if I am wrong, but this class is just to have our custom xdoc template files. It is planned to be removed as part of the #13426 at which time we will be using pure doxia classes. When we remove the custom code, we will also be removing the override.
> 
> I recommend we create an issue for doxia, especially if we see a way they can allow an override.

@rnveach - exactly I was meaning this!

Reply:
https://github.com/checkstyle/checkstyle/blob/d94f78fffde2b3180b911c6d8f8c79250803cb9d/src/main/java/com/puppycrawl/tools/checkstyle/site/XdocsTemplateSink.java#L38

https://github.com/apache/maven-doxia/blob/14eaf4268135ee90f400bc929d32052089f900a6/doxia-modules/doxia-module-xdoc/src/main/java/org/apache/maven/doxia/module/xdoc/XdocSink.java#L42

https://github.com/apache/maven-doxia/blob/14eaf4268135ee90f400bc929d32052089f900a6/doxia-core/src/main/java/org/apache/maven/doxia/sink/impl/Xhtml5BaseSink.java#L1999-2008

Reply:
https://github.com/apache/maven-doxia/blob/14eaf4268135ee90f400bc929d32052089f900a6/doxia-modules/doxia-module-xdoc/src/main/java/org/apache/maven/doxia/module/xdoc/XdocSink.java#L42

Reply:
I don't see XhtmlBaseSink. And look like we can override protected method.

Reply:
@romani 
if I override the method like this in `XdocsTemplateSink` then certain variables becomes undefined.
Like `witer` , `tableCaptionXMLWriterStack` ... etc are private variables of `XhtmlBaseSink` class or `Xhtml5BaseSink` , hence I am forbidden to use the `super` keyword too.

```
@Override
    protected void write( String text ) {
        if ( !this.tableCaptionXMLWriterStack.isEmpty() && this.tableCaptionXMLWriterStack.getLast() != null ) {
            this.tableCaptionXMLWriterStack.getLast().writeMarkup( "\n" );
        }
        else if ( !this.tableContentWriterStack.isEmpty() && this.tableContentWriterStack.getLast() != null ) {
            this.tableContentWriterStack.getLast().write( "\n" );
        }
        else {
            writer.write( "\n" );
        }
    }
```

Reply:
Please store writer at class field in  ctor https://github.com/checkstyle/checkstyle/blob/d94f78fffde2b3180b911c6d8f8c79250803cb9d/src/main/java/com/puppycrawl/tools/checkstyle/site/XdocsTemplateSink.java#L49

Reply:
I do not see this to be done or tried.